Running on host gpu32.bc4.acrc.priv
Slurm job ID is 3498457
This job runs on the following machines:
gpu32
Requirement already satisfied: matplotlib in /mnt/storage/home/mk17486/.local/lib/python3.7/site-packages (3.2.1)
Requirement already satisfied: kiwisolver>=1.0.1 in /mnt/storage/home/mk17486/.local/lib/python3.7/site-packages (from matplotlib) (1.1.0)
Requirement already satisfied: pyparsing!=2.0.4,!=2.1.2,!=2.1.6,>=2.0.1 in /mnt/storage/home/mk17486/.local/lib/python3.7/site-packages (from matplotlib) (2.4.6)
Requirement already satisfied: cycler>=0.10 in /mnt/storage/home/mk17486/.local/lib/python3.7/site-packages (from matplotlib) (0.10.0)
Requirement already satisfied: numpy>=1.11 in /mnt/storage/home/mk17486/.local/lib/python3.7/site-packages (from matplotlib) (1.18.1)
Requirement already satisfied: python-dateutil>=2.1 in /mnt/storage/home/mk17486/.local/lib/python3.7/site-packages (from matplotlib) (2.8.1)
Requirement already satisfied: setuptools in /mnt/storage/software/languages/miniconda/miniconda3/lib/python3.7/site-packages (from kiwisolver>=1.0.1->matplotlib) (41.0.1)
Requirement already satisfied: six in /mnt/storage/software/languages/miniconda/miniconda3/lib/python3.7/site-packages (from cycler>=0.10->matplotlib) (1.12.0)
Requirement already satisfied: torchvision in /mnt/storage/home/mk17486/.local/lib/python3.7/site-packages (0.5.0)
Requirement already satisfied: six in /mnt/storage/software/languages/miniconda/miniconda3/lib/python3.7/site-packages (from torchvision) (1.12.0)
Requirement already satisfied: torch==1.4.0 in /mnt/storage/home/mk17486/.local/lib/python3.7/site-packages (from torchvision) (1.4.0)
Requirement already satisfied: numpy in /mnt/storage/home/mk17486/.local/lib/python3.7/site-packages (from torchvision) (1.18.1)
Requirement already satisfied: pillow>=4.1.1 in /mnt/storage/home/mk17486/.local/lib/python3.7/site-packages (from torchvision) (7.0.0)
Requirement already satisfied: ninja in /mnt/storage/home/mk17486/.local/lib/python3.7/site-packages (1.9.0.post1)
Processing /mnt/storage/home/mk17486/neural-networks/pytorch-subLSTM
Requirement already satisfied: numpy in /mnt/storage/home/mk17486/.local/lib/python3.7/site-packages (from torch-sublstm==0.1.0) (1.18.1)
Requirement already satisfied: torch in /mnt/storage/home/mk17486/.local/lib/python3.7/site-packages (from torch-sublstm==0.1.0) (1.4.0)
Building wheels for collected packages: torch-sublstm
  Building wheel for torch-sublstm (setup.py): started
  Building wheel for torch-sublstm (setup.py): still running...
  Building wheel for torch-sublstm (setup.py): finished with status 'done'
  Stored in directory: /mnt/storage/home/mk17486/.cache/pip/wheels/31/17/41/f7b13d3b73f5822baeb1d95f85487aba02a62f36c7b48952c9
Successfully built torch-sublstm
Installing collected packages: torch-sublstm
  Found existing installation: torch-sublstm 0.1.0
    Uninstalling torch-sublstm-0.1.0:
      Successfully uninstalled torch-sublstm-0.1.0
Successfully installed torch-sublstm-0.1.0
cuda is available: True
cudnn enabled: True
Training LSTM model with parameters:
	number of layers: 1
	hidden units: 50
	max epochs: 1
	batch size: 50
	optimizer: rmsprop, lr=0.0001, l2=0
	using CUDA
training epoch 1
Traceback (most recent call last):
  File "run.py", line 225, in <module>
    verbose=args.verbose
  File "../utils.py", line 93, in train
    model.rnn.totalbackwardtime += btime
  File "/mnt/storage/home/mk17486/.local/lib/python3.7/site-packages/torch/nn/modules/module.py", line 576, in __getattr__
    type(self).__name__, name))
AttributeError: 'LSTM' object has no attribute 'totalbackwardtime'
cuda is available: True
cudnn enabled: True
Training subLSTMCuda model with parameters:
	number of layers: 1
	hidden units: 50
	max epochs: 1
	batch size: 50
	optimizer: rmsprop, lr=0.0001, l2=0
	using CUDA
training epoch 1
	[batches    20 /   960] loss: 2.30801
	[batches    40 /   960] loss: 2.29933
	[batches    60 /   960] loss: 2.30279
	[batches    80 /   960] loss: 2.30224
	[batches   100 /   960] loss: 2.30433
	[batches   120 /   960] loss: 2.30403
	[batches   140 /   960] loss: 2.30317
	[batches   160 /   960] loss: 2.30220
	[batches   180 /   960] loss: 2.30101
	[batches   200 /   960] loss: 2.30440
	[batches   220 /   960] loss: 2.30307
	[batches   240 /   960] loss: 2.30188
	[batches   260 /   960] loss: 2.30456
	[batches   280 /   960] loss: 2.30220
	[batches   300 /   960] loss: 2.29884
	[batches   320 /   960] loss: 2.30214
	[batches   340 /   960] loss: 2.29948
	[batches   360 /   960] loss: 2.29916
	[batches   380 /   960] loss: 2.30201
	[batches   400 /   960] loss: 2.30599
	[batches   420 /   960] loss: 2.30357
	[batches   440 /   960] loss: 2.30107
	[batches   460 /   960] loss: 2.30396
	[batches   480 /   960] loss: 2.29979
	[batches   500 /   960] loss: 2.30158
	[batches   520 /   960] loss: 2.29960
	[batches   540 /   960] loss: 2.30199
	[batches   560 /   960] loss: 2.30051
	[batches   580 /   960] loss: 2.30502
	[batches   600 /   960] loss: 2.29986
	[batches   620 /   960] loss: 2.30051
	[batches   640 /   960] loss: 2.30067
	[batches   660 /   960] loss: 2.29735
	[batches   680 /   960] loss: 2.30283
	[batches   700 /   960] loss: 2.30253
	[batches   720 /   960] loss: 2.30295
	[batches   740 /   960] loss: 2.30277
	[batches   760 /   960] loss: 2.30350
	[batches   780 /   960] loss: 2.30066
	[batches   800 /   960] loss: 2.30097
	[batches   820 /   960] loss: 2.30192
	[batches   840 /   960] loss: 2.30121
	[batches   860 /   960] loss: 2.30454
	[batches   880 /   960] loss: 2.29891
	[batches   900 /   960] loss: 2.30474
	[batches   920 /   960] loss: 2.30212
	[batches   940 /   960] loss: 2.30043
	[batches   960 /   960] loss: 2.30362
epoch 1 finished                 
	total time 778.1167371273041                 
	training_loss = 2.3021                 
	validation_loss = 2.3023
total time to train 778.1167371273041
Traceback (most recent call last):
  File "run.py", line 268, in <module>
    if (model_type != 'LSTM'):
NameError: name 'model_type' is not defined
